add Pocket and Instapaper share buttons and icons

jwrubel opened the pull request:
Hi! I needed these two services for a 'read later' function so I added them. I've checked the demo page and they look ok to me. I used PR #181 as a reference here - Let me know if I missed anything.

Cheers, and thanks for making this awesome library available!

jwrubel commented:
this is the result of running the demos for me, with the two new icons at the end 
<img width="513" alt="image 2018-12-18 at 19 54 22" src="https://user-images.githubusercontent.com/1226418/50192113-fb258680-02fe-11e9-85a4-3b21730b3f48.png">

aautio commented on src/InstapaperShareButton.js:
```diff
@@ -0,0 +1,26 @@
+
+  return 'http://www.instapaper.com/hello2' + objectToGetParams({
+    url,
+    text: title,
```
Should the name of the parameter be `title`? 

I did not find the official documentation, but https://feedbin.com/help/sharing-read-it-later-services/ and https://gist.github.com/revelt/a70923df337f7923e067 suggest that `url` and `title` would be the parameters to use here.

jwrubel replied:
I found [the documentation](https://www.instapaper.com/publishers) and `title` is correct, plus it allows a `description` property. I'll add these.

aautio commented on src/PocketShareButton.js:
```diff
@@ -0,0 +1,26 @@
+
+  return 'https://getpocket.com/save' + objectToGetParams({
+    url,
+    text: title,
```
The same question over here: should the query parameter be named `title`? The links mentioned in the previous comment provide examples for Pocket. There it seems that `title` is used.

jwrubel replied:
Good catch! After reviewing the [documentation for their API](https://getpocket.com/developer/docs/v3/add) it looks like the title parameter is ignored if Pocket can get a title parameter from the shared page, so it looked like it was working for me when I tested. I'll make this change, and also I notice that Pocket supports a `tags` property and also a `tweet_id` so I'll update to add those.

jwrubel commented:
okay @aautio I've updated the code to reflect `title` as the parameter name for both services and added a `description` property for Instapaper. After some research, it appears that the bookmarklet link we're using for pocket here doesn't accept ` tags` or `tweet_id` if they are passed. Only the Pocket API allows this.

xuopled commented:
It will be awesome when this PR will be merged 🎉

aautio commented:
Thank you @jwrubel 👍 

You can follow #210 to see when the Pocket and Instapaper -additions are published to npm.
